libkbfs/folder_branch_ops.go

```diff
@@ -3876,13 +3923,16 @@ func (fbo *folderBranchOps) syncAllLocked(
 			return err
 		})
 
-		// Make each of the ops "self-updates" before making chains
-		// (they will be updated later to have the correct block
-		// pointers).  Add an update for each element in the path, so
-		// that the chains below are created correctly.
+		// Add an "update" for all the file update, and make chains
+		// for the rest of the parent directories, so they're treated
+		// like updates during the prepping.
 		lastOp := md.Data().Changes.Ops[len(md.Data().Changes.Ops)-1]
-		for _, p := range file.path {
-			lastOp.AddUpdate(p.BlockPointer, p.BlockPointer)
+		for i, pn := range file.path {
+			if i == len(file.path)-1 {
+				lastOp.AddUpdate(pn.BlockPointer, pn.BlockPointer)
+			} else {
+				parentsToAddChainsFor[pn.BlockPointer] = true
+			}
```

Comment:
This block is the same as https://github.com/keybase/kbfs/pull/952/files#diff-f6470cb87de71869397b1e97dd1ee43dR3776. Helper function?